pkg/controller/baremetalhost/action_result.go

```diff
@@ -90,11 +95,22 @@ func (r actionError) Dirty() bool {
 // actionFailed is a result indicating that the current action has failed,
 // and that the resource should be marked as in error.
 type actionFailed struct {
-	dirty     bool
-	ErrorType metal3.ErrorType
+	dirty      bool
+	ErrorType  metal3.ErrorType
+	errorCount int
+}
+
+func calculateBackoff(errorCount int, max time.Duration) time.Duration {
+	backOff := math.Exp2(float64(errorCount))
+	backOffDuration := time.Second * time.Duration(backOff)
+	if backOffDuration.Milliseconds() > max.Milliseconds() {
+		return max
+	}
+	return backOffDuration
```

Comment:
Having fixed delays (even exponentially increasing ones) is prone to causing thundering herd problems. We should at least add some jitter on top. (We could go as far as to implement exponential backoff in the CSMA sense, where we wait for a random interval between 0 and backOffDuration, but resource contention isn't our primary reason for backing off here so my instinct is that that would be overkill.)

Reply:
We have a separate need to throttle the number of things we ask ironic to do at once anyway, so maybe we can solve the herd problem that way to avoid complicating the logic here?